x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizer.java

```diff
@@ -422,4 +441,16 @@ public Query wrapFunctionQuery(ScalarFunction sf, Expression field, Supplier<Que
             throw new EsqlIllegalArgumentException("Expected a FieldAttribute or MetadataAttribute but received [" + field + "]");
         }
     }
+
+    public static boolean hasIdenticalDelegate(FieldAttribute attr, SearchStats stats) {
+        return stats.hasIdenticalDelegate(attr.name());
+    }
+
+    public static boolean isPushableFieldAttribute(Expression exp, Function<FieldAttribute, Boolean> hasIdenticalDelegate) {
+        if (exp instanceof FieldAttribute fa && fa.getExactInfo().hasExact() && isAggregatable(fa)) {
+            return fa.dataType().equals(DataTypes.TEXT) ? hasIdenticalDelegate.apply(fa) : true;
```

Comment:
`return fa.dataType().equals(DataTypes.TEXT) && hasIdenticalDelegate.test(fa)`

Reply:
I think that wouldn't be right for non-TEXT, but `return fa.dataType() != DataTypes.TEXT || hasIdenticalDelegate.test(fa)` should be. 
Or `return exp instanceof FieldAttribute fa && fa.getExactInfo().hasExact() && isAggregatable(fa) && (fa.dataType() != DataTypes.TEXT || hasIdenticalDelegate.test(fa));`.